src/rules/commentFormatRule.ts

```diff
@@ -178,27 +201,78 @@ function walk(ctx: Lint.WalkContext<Options>) {
             return;
         }
 
+        const failures: Failures = { leadingSpace: false, uppercase: false, lowercase: false, firstLetterPos: -1};
         if (ctx.options.space && commentText[0] !== " ") {
-            ctx.addFailure(start, end, Rule.LEADING_SPACE_FAILURE);
+            failures.leadingSpace = true;
         }
 
         if (ctx.options.case === Case.None ||
             ctx.options.exceptions !== undefined && ctx.options.exceptions.test(commentText) ||
             ENABLE_DISABLE_REGEX.test(commentText)) {
+            addFailure(ctx, commentText, start, end, failures);
             return;
         }
 
         // search for first non-space character to check if lower or upper
         const charPos = commentText.search(/\S/);
         if (charPos === -1) {
+            addFailure(ctx, commentText, start, end, failures);
             return;
         }
-        if (ctx.options.case === Case.Lower) {
-            if (!isLowerCase(commentText[charPos])) {
-                ctx.addFailure(start, end, Rule.LOWERCASE_FAILURE + ctx.options.failureSuffix);
+        failures.firstLetterPos = charPos;
+        if (ctx.options.case === Case.Lower && !isLowerCase(commentText[charPos])) {
+            failures.lowercase = true;
+        } else if (ctx.options.case === Case.Upper && !isUpperCase(commentText[charPos])) {
+            failures.uppercase = true;
+            if (ctx.options.allowTrailingLowercase) {
+                const lineAndCharCurrentComment = ts.getLineAndCharacterOfPosition(ctx.sourceFile, pos);
+                const posPrevComment = ctx.sourceFile.getPositionOfLineAndCharacter(
+                    lineAndCharCurrentComment.line - 1, 0) + lineAndCharCurrentComment.character;
+
+                const prevComment = utils.getCommentAtPosition(ctx.sourceFile, posPrevComment);
```

Comment:
instead of the lookbehind on every comment, you could just save the position of the last seen comment in a variable to reuse it on the next comment